network/socket-bench-server/src/lib.rs

```diff
@@ -76,21 +76,31 @@ impl Args {
 }
 
 /// Build a MemorySocket + Noise transport
-pub fn build_memsocket_noise_transport() -> impl Transport<Output = NoiseSocket<MemorySocket>> {
+pub fn build_memsocket_noise_transport() -> impl Transport<Output = NoiseSession<MemorySocket>> {
     MemoryTransport::default().and_then(move |socket, _addr, origin| async move {
         let mut rng: StdRng = SeedableRng::from_seed(TEST_SEED);
-        let noise_config = Arc::new(NoiseConfig::new_random(&mut rng));
-        let (_remote_static_key, socket) = noise_config.upgrade_connection(socket, origin).await?;
+        let private = x25519::PrivateKey::generate(&mut rng);
+        let dummy_public = private.public_key();
+        let noise_config = Arc::new(NoiseWrapper::new(private));
+        // TODO: get this from addr
+        let (_remote_static_key, socket) = noise_config
+            .upgrade_connection(socket, origin, dummy_public, None)
+            .await?;
         Ok(socket)
     })
 }
 
 /// Build a Tcp + Noise transport
-pub fn build_tcp_noise_transport() -> impl Transport<Output = NoiseSocket<TcpSocket>> {
+pub fn build_tcp_noise_transport() -> impl Transport<Output = NoiseSession<TcpSocket>> {
     TcpTransport::default().and_then(move |socket, _addr, origin| async move {
         let mut rng: StdRng = SeedableRng::from_seed(TEST_SEED);
-        let noise_config = Arc::new(NoiseConfig::new_random(&mut rng));
-        let (_remote_static_key, socket) = noise_config.upgrade_connection(socket, origin).await?;
+        let private = x25519::PrivateKey::generate(&mut rng);
+        let dummy_public = private.public_key();
+        let noise_config = Arc::new(NoiseWrapper::new(private));
+        // TODO: get this from addr
```

Comment:
where is the address coming from, can we actually trust it?

Reply:
that's a bencher : o

Reply:
I guess I still don't understand... there's no address here, so how do we intend on getting this from an address? I could imagine extracting an address from the outgoing socket... but I don't think that API really exists. More importantly, do we really care about extracting "addresses" from sockets?

Reply:
updated this part, like other parts of the code it now returns None if there's no public key, and only errors if there is no public key AND we are dialing

```
let remote_public_key = addr.find_noise_proto();
```